src/runtime/resolver.rs

```diff
@@ -0,0 +1,160 @@
+use std::{future::Future, net::SocketAddr, time::Duration};
+
+use async_trait::async_trait;
+use futures_timer::Delay;
+use trust_dns_proto::error::ProtoError;
+use trust_dns_resolver::{
+    lookup::{SrvLookup, TxtLookup},
+    name_server::{GenericConnection, GenericConnectionProvider},
+    IntoName,
+};
+
+use super::stream::AsyncTcpStream;
+use crate::{
+    error::{Error, Result},
+    runtime::AsyncRuntime,
+    RUNTIME,
+};
+
+// Some shorter type names for the generic constraints of the underlying trust-dns
+// `AsyncResolver`.
+type AsyncConnectionProvider = GenericConnectionProvider<AsyncRuntimeProvider>;
+type TrustDnsResolver =
+    trust_dns_resolver::AsyncResolver<GenericConnection, AsyncConnectionProvider>;
```

Comment:
trust-dns already has built in support for tokio, but since I had to implement all of this anyways for async-std, it was easier to not conditionally use the built in support and to just have a single type that relies on our agnostic runtime.